core/domain/topic_domain_test.py

```diff
@@ -821,6 +821,497 @@ def test_cannot_move_existing_skill_to_subtopic(self) -> None:
+            topic_domain.Subtopic(
+                2, 'Another title', ['skill_id_2'], 'image.svg',
+                constants.ALLOWED_THUMBNAIL_BG_COLORS['subtopic'][0], 21131,
+                'dummy-subtopic-two')]
+        self.topic.move_skill_id_to_subtopic(1, 2, 'skill_id_1')
+        self.assertEqual(
+            self.topic.subtopics[0].skill_ids,
+            expected_subtopic1_skills
+        )
+        self.assertEqual(
+            self.topic.subtopics[1].skill_ids,
+            expected_subtopic2_skills
+        )
+
+    def test_move_skill_id_from_uncategorized_to_subtopic(self) -> None:
+        """Checks that move_skill_id_to_subtopic works when moving a skill_id
+        from an existing subtopic to a new subtopic returns the expected
+        updated values for skill_ids associated with each subtopic.
+        """
+        expected_subtopic_skills = ['skill_id_2', 'skill_id_3']
+        expected_uncategorized_skills: list[str] = []
+        self.topic.uncategorized_skill_ids = ['skill_id_3']
+        self.topic.subtopics = [
+            topic_domain.Subtopic(
+                1, 'Title', ['skill_id_1'], 'image.svg',
+                constants.ALLOWED_THUMBNAIL_BG_COLORS['subtopic'][0], 21131,
+                'dummy-subtopic-one'),
+            topic_domain.Subtopic(
+                2, 'Another title', ['skill_id_2'], 'image.svg',
+                constants.ALLOWED_THUMBNAIL_BG_COLORS['subtopic'][0], 21131,
+                'dummy-subtopic-two')]
+        self.topic.move_skill_id_to_subtopic(None, 2, 'skill_id_3')
+        self.assertEqual(
+            self.topic.subtopics[1].skill_ids,
+            expected_subtopic_skills
+        )
+        self.assertEqual(
+            self.topic.uncategorized_skill_ids,
+            expected_uncategorized_skills
+        )
+
+    def test_add_subtopic(self) -> None:
+        """Checkts that if next_subtopic_id isn't correct
+        an exception is raised. Also checks for the sub topic
+        getting added to the topic.
+        """
+        incorrect_new_subtopic_id = 3
+        correct_new_subtopic_id = 2
+        expected_subtopic_id = self.topic.next_subtopic_id
+        with self.assertRaisesRegex(
+            Exception,
+            'The given new subtopic id %s is not equal to the expected next '
+            'subtopic id: %s' % (
+                incorrect_new_subtopic_id,
+                expected_subtopic_id
+            )
+        ):
+            self.topic.add_subtopic(
+                incorrect_new_subtopic_id,
+                'subtopic_3',
+                'url_frag'
+            )
+        self.topic.add_subtopic(
+            correct_new_subtopic_id,
+            'subtopic_title',
+            'url_frag'
+            )
+        self.assertEqual(2, len(self.topic.subtopics))
+
+    def test_update_practice_tab_is_displayed(self) -> None:
+        self.topic.update_practice_tab_is_displayed(True)
+        self.assertTrue(self.topic.practice_tab_is_displayed)
```

Comment:
For tests like this, please add pre-checks to make sure that the condition you assert at the end is not true at the beginning. In this case, that means adding `self.assertFlase(self.topic.practice_tab_is_displayed)` to the start of the test. See point 2 here: https://github.com/oppia/oppia/wiki/Backend-tests#guidelines-for-writing-good-tests. Ditto below

Reply:
I believe I found them all and made the recommended changes. Thanks for the feedback, now that I have a little experience writing these I think it would be good to go through that link again with a little more context.
Done.